Rename `Card` module to `CardPayments`

jcnoriega opened the pull request:
### Reason for changes
We have to rename to `CardPayments` because it better reflects the functionality of the module


### Summary of changes

- Rename Target and folder
- Updated podspec
- Rename Unit test folder as well
- Updated `Package.swift`
- Update docs

### Checklist

- [x] Added a changelog entry

### Authors
> List GitHub usernames for everyone who contributed to this pull request.

- @jcnoriega

mattwylder commented on CHANGELOG.md:
```diff
@@ -2,8 +2,10 @@
+* Rename `Card` to `CardPayments`
 * PayPalUI
     * Fix issue where label was not being shown
+>>>>>>> main
```
looks like some merge conflict text got caught in commit.

jaxdesmarais reviewed (commented):
Looks like a just few minor places got missed:
1. Updating the bundle ID for `Card` and `CardTests` (under signing & capabilities > bundle ID)
2. The above _should_ update the other build settings, but worth searching the "Build Settings" tab for "card" or "cardtests" after making these updates to be sure
3. `CardUITests` > `CardPaymentsUITests` - will also need to update this file name
4. Should we update the `Card` demo setting to `Card Payments`? Not sure if this makes sense but worth looking into
5. Strings like "Buyer has cancelled the Card flow" should probably be updated to "Buyer has cancelled the Card Payments flow"
6. Should we update `CardDemoViewController` and `SwiftUICardDemo` to card payments?

Great work so far on this! Renaming module is a pain in the 🍑  and these are all _super minor_ comparatively.

jcnoriega commented:
@jaxdesmarais thank you for pointing that out! yeah i wish with just refactoring the name, that xcode would me smart enough to change everything lol.
A couple of points:
1. I updated the bundle id on every PR.
2. I was planning on doing a follow up PR where i updated some minor stuff on the Demo app. I didnt want to overload these PRs with to much. (i also have to merge cascade with all the other branches, as the PRs depend on one another)
3. I would for now leave the doc strings as they are, because we only changed the module name, the classes remaing the same (specially with `CardDelegate` that all of its methods are `card`).  Maybe after we decide on a diferent naming convention  like we talked about in [this Slack thread](https://braintree.slack.com/archives/C03U9SEQYHK/p1674588560726189), we can talk update

jaxdesmarais commented:
> @jaxdesmarais thank you for pointing that out! yeah i wish with just refactoring the name, that xcode would me smart enough to change everything lol. A couple of points:
> 
> 1. I updated the bundle id on every PR.
> 2. I was planning on doing a follow up PR where i updated some minor stuff on the Demo app. I didnt want to overload these PRs with to much. (i also have to merge cascade with all the other branches, as the PRs depend on one another)
> 3. I would for now leave the doc strings as they are, because we only changed the module name, the classes remaing the same (specially with `CardDelegate` that all of its methods are `card`).  Maybe after we decide on a diferent naming convention  like we talked about in [this Slack thread](https://braintree.slack.com/archives/C03U9SEQYHK/p1674588560726189), we can talk update

Thanks so much for the detailed reply (and making those updates)! I think punting on a few of these items sounds good while we wait to solidify some other naming updates.

jaxdesmarais commented on PayPal.xcodeproj/project.pbxproj:
```diff
@@ -2359,7 +2359,7 @@
 				MTL_ENABLE_DEBUG_INFO = NO;
 				MTL_FAST_MATH = YES;
-				PRODUCT_BUNDLE_IDENTIFIER = "com.paypal.ios-sdk.Card";
+				PRODUCT_BUNDLE_IDENTIFIER = "com.paypal.ios-sdk.CardPayments";
```
It looks like the debug bundle ID is still using `Card`:
<img width="891" alt="Screen Shot 2023-01-30 at 1 17 04 PM" src="https://user-images.githubusercontent.com/20733831/215573882-d68e0778-1a5b-458e-a813-a9270841f722.png">

jaxdesmarais replied:
Same for `CardPaymentTests` though I imagine we want this to be `CardPaymentsTests` in both places instead of the singular `Payment`.

jcnoriega replied:
i'll check again if i missed another spot, but is there a way for you to do a "clean"? In Android Studio sometimes we have to do an "Invalidate Cache" to really test stuff that affect the build. Is there an equivalent in xcode?

jcnoriega replied:
@jaxdesmarais never mind!
i found were it was missing, thank you!

jaxdesmarais commented on SampleApps/SPMTest/SPMTest.xcodeproj/project.pbxproj:
```diff
@@ -368,22 +368,22 @@
 			repositoryURL = "https://github.com/paypal/iOS-SDK.git";
 			requirement = {
-				branch = paypal_web_integration;
+				branch = "rename-card";
```
Just want to make sure, do we have a ticket to point this back to `main` once all of the renaming work is completed? Want to make sure it doesn't get missed.

jcnoriega replied:
@jaxdesmarais i dont have a ticket for it, but i was planning in fixing this later, thats why in the other module renaming i commented the GithubActions "SPMTest", it was creating all sort of problems with the merge, but ill write a ticket for it.

jcnoriega replied:
@jaxdesmarais ticket here: https://paypal.atlassian.net/browse/DTNOR-810

jaxdesmarais reviewed (approved):
👻
